modules/javafx.graphics/src/main/native-glass/gtk/GlassApplication.cpp

```diff
@@ -447,7 +447,7 @@ static void process_events(GdkEvent* event, gpointer data)
         return;
     }
 
-    if (ctx != NULL && ctx->hasIME() && ctx->filterIME(event)) {
+    if (event->type == GDK_KEY_PRESS && ctx != NULL && ctx->filterIME(event)) {
```

Comment:
In the previous code both PRESS and RELEASE events were sent to the filter and gtk_im_context_filter_keypress says that it can filter both. Is there some particular reason you're not calling this on RELEASE events?

Reply:
It seems to have no effect on RELEASE, but as the documentation states [press and realease](https://docs.gtk.org/gtk3/method.IMContext.filter_keypress.html) I've restored it.